concordium-consensus/src/Concordium/GlobalState/Persistent/BlockState.hs

```diff
@@ -108,6 +109,33 @@ data PersistentBirkParameters (av :: AccountVersion) = PersistentBirkParameters
 
 makeLenses ''PersistentBirkParameters
 
+migratePersistentBirkParameters :: forall oldpv pv t m .
+    ( IsProtocolVersion pv
+    , IsProtocolVersion oldpv
+    , SupportMigration m t
+    , SupportsPersistentAccount pv (t m)
+    ) =>
+  StateMigrationParameters oldpv pv ->
+  Accounts.Accounts pv ->
+  PersistentBirkParameters (AccountVersionFor oldpv) ->
+  t m (PersistentBirkParameters (AccountVersionFor pv))
+migratePersistentBirkParameters migration accounts PersistentBirkParameters{..} = do
+  bakerIds <- lift (Trie.keysAsc . _activeBakers =<< refLoad _birkActiveBakers)
+  totalBakerStakedAmount <- foldM (\acc (BakerId aid) ->
+    Accounts.indexedAccount aid accounts >>= \case
+      Nothing -> error "Baker account does not exist."
+      Just pa -> return $! (acc + _accountAmount pa)) 0 bakerIds
```

Comment:
```suggestion
  totalBakerStakedAmount <- foldM (\acc (BakerId aid) ->
    Accounts.indexedAccount aid accounts >>= \case
      Nothing -> error "Baker account does not exist."
      Just pa -> case pa ^. accountBaker of
        Null -> error "Baker account does not exist."
        Some bref -> do
            bkr <- refLoad bref
            return $! (acc + _stakedAmount bkr)) 0 bakerIds
```

Reply:
Good catch. I guess this was not caught in the testing because this structure is not hashed.

I'm still a bit surprised that this did not affect at all the behaviour of the node on mainnet and testnet.

Reply:
Also moved the implementation to `migratePersistentActiveBakers` as suggested.